std/zip.d

```diff
@@ -410,19 +412,26 @@ final class ZipArchive
      *
      * Returns: array representing the entire archive.
      */
-    void[] build()
+    void[] build() @safe pure
     {
         import std.algorithm.sorting : sort;
+        import std.string : representation;
+
         uint i;
         uint directoryOffset;
 
         if (comment.length > 0xFFFF)
             throw new ZipException("archive comment longer than 65535");
 
+        static auto vals(T)(T a) @trusted
+        {
+            return a.values().sort!((x, y) => x.index < y.index).release;
```

Comment:
Here is the only use of `@trusted` in the PR. Why is this sorting not considered `safe` by the compiler, and why is it known to humans to be memory-safe?